### PR TITLE
[RFC] Make "v:errmsg", "v:shell_error" and "v:this_session" distinct

### PR DESCRIPTION
**Pull request by Shougo**
It is the next of https://github.com/neovim/neovim/pull/7407.

"v:errmsg", "v:shell_error" and "v:this_session" distinct from `v:` variables.

Note:  "v:version" is not changed yet.
The next PR is for "v:version" variable and to remove `VV_COMPAT` feature.

### CODE REVIEW

**Comment by justinmk** [comment]
We can do this for v:errmsg in 0.3.1, but the others are too risky for 0.3.1, which will be released in a few weeks.

> Note: "v:version" is not changed yet.
> The next PR is for "v:version" variable and to remove VV_COMPAT feature.

I don't think we can ever do that. It will break many plugins which blindly copy the `version` check from old examples.

**Comment by Shougo** [comment]
> We can do this for v:errmsg in 0.3.1, but the others are too risky for 0.3.1, which will be released in a few weeks.

I don't know shell_error or this_session is used in the plugins.
I agree with it is risky for 0.3.1.

> I don't think we can ever do that. It will break many plugins which blindly copy the version check from old examples.

Any examples exists?

**Comment by Shougo** [comment]
OH, it is too bad.

https://github.com/search?l=Vim+script&q=version+user%3Avim-scripts&type=Code

`version` is used for `v:version` in many old Vim scripts.

**Comment by justinmk** [comment]
Yes, even Vim runtime has at least 17 examples. https://github.com/vim/vim/blob/c8523e2e6cd072d86a9412f465aa9eef53f5675e/runtime/autoload/ada.vim#L31

**Comment by Shougo** [comment]
`session` is used for `v:version`, it seems bug.

https://github.com/vim-scripts/session.vim/blob/6bd8becf666d81a17c9b7e5e8478dad23e7345f7/plugin/session.vim

So `v:this_session` is not high risk.

**Comment by justinmk** [comment]
? In that script you linked, `this_session` is assigned: https://github.com/vim-scripts/session.vim/blob/6bd8becf666d81a17c9b7e5e8478dad23e7345f7/plugin/session.vim#L63

And that's how `v:this_session` is used. So that plugin would break.

**Comment by Shougo** [comment]
> And that's how v:this_session is used. So that plugin would break.

Sorry, it is in `if_python` routine.
So it does not use `this_session` variable in Vim script.
It works with this PR.

I have researched `this_session` and `shell_error` in vim-scripts.org.
They are not used in old Vim scripts.  So it is no risk.

**Comment by justinmk** [comment]
My mistake. Let's do this after the 0.3.1 release,

**Comment by Shougo** [comment]
Hi.  0.3.1 is merged.  It will be merged in 0.3.2?

File: runtime/doc/vim_diff.txt [hunk 1]
```diff
@@ -349,7 +349,10 @@ TUI:
   uses one or the other, it does not attempt to merge the two.
 
 VimL (Vim script) compatibility:
-  `count` does not alias to |v:count|
+  `count` do not alias to |v:count|
+  `errmsg` do not alias to |v:errmsg|
+  `shell_error` do not alias to |v:shell_error|
+  `this_session` do not alias to |v:this_session|
```
Comment (justinmk) [inline runtime/doc/vim_diff.txt]: "does not" is probably better for all of these cases.

File: runtime/doc/eval.txt [hunk 2]
```diff
@@ -1813,7 +1812,6 @@ v:shell_error	Result of the last shell command.  When non-zero, the last
 	:if v:shell_error
 	:  echo 'could not rename "foo" to "bar"!'
 	:endif
-<		"shell_error" also works, for backwards compatibility.
```
Comment (justinmk) [inline runtime/doc/eval.txt]: I think the `<` in column 1 is still needed .

**Comment by justinmk** [comment]
@Shougo If you can rebase and address the minor comments , will merge.

**Comment by Shougo** [comment]
Thanks.  I have fixed the problems and rebased.

File: runtime/doc/eval.txt (hunk 2 above)
Comment (justinmk) [reply runtime/doc/eval.txt]: The other case as well. I fixed this in the merge-commit.

**Comment by Shougo** [comment]
Thank you

**Pull request merged**